plugin/pkg/scheduler/schedulercache/interface.go

```diff
@@ -0,0 +1,81 @@
+// Pod events are sent via network. We don't have guaranteed delivery of all events except last seen.
+// Thus, we organized the state machine flow of a pod's events and handle it clearly.
+//
+// State Machine (life cycle) of a pod in scheduler's cache:
+//
+//                                                +-------+
+//                                                |       |
+//                                                |       | Update
+//           Assume                Add            +       |
+// Initial +--------> Assumed +----------------> Added <--+
+//                      +                         +
+//                      |                         |
+//                      |                         |
+//                      | Remove                  | Remove
+//                      |                         v
+//                      +------------------->  Deleted
+//
+// Depending on the implementation, it might choose to manage the expiration of assumed pod.
+// Thus, a new state "Expired" could be added to the diagram:
+//
+//                                                     +--------+
+//                                                     |        |
+//                                                     |        | Update
+//                                       Add           +        |
+//                        +---------------+------->  Added <----+
+//                        |               |            +
+//                        |               |            |
+//                        |               |            |
+//           Assume       +    expire     +            |
+// Initial +-------> Assumed +--------> Expired        |
+//                        +               +            |Remove
+//                        |               |            |
+//                        |               |            v
+//                        +---------------+------> Deleted
+//                                      Remove
+//
+// Note:
+// - Both "Initial" and "Deleted" pods do not actually exist in cache.
+//   In order to differentiate them, we need external request to guarantee
+//   no same pod will be created twice.
+type NodeInfoCache interface {
+	// GetNodeInfo returns aggregated node information for given node name.
+	// If no pod's been added on the node, it returns nil.
+	// For better concurrency control, we use a callback to serve the result.
+	// Depending on the implementation, it might acquire a lock. So don't do anything time consuming inside.
+	GetNodeInfo(nodeName string, callback func(*NodeInfo))
+	// AssumePod assumes a pod to be scheduled. The pod's information is aggregated into assigned node.
+	// The implementation might decide the policy to expire/remove the assumed pod before it is confirmed to be scheduled.
+	// After expiration, its information would be subtracted.
+	AssumePod(pod *api.Pod) error
```

Comment:
Assumed is a better state term to differentiate "assumed to be scheduled" and "confirmed it's scheduled".

If it's confusing, how about `AssumePodScheduled`?